File: src/main/java/gregtechlite/gtlitecore/api/network/NetworkHandler.java
package gregtechlite.gtlitecore.api.network;

import net.minecraft.entity.Entity;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraftforge.fml.common.network.NetworkRegistry;

public interface NetworkHandler
{

    void registerPacket(Class<? extends NetworkPacket> packetClass);

    void sendToAll(NetworkPacket packet);

    void sendTo(NetworkPacket packet, EntityPlayerMP player);

    void sendToAllAround(NetworkPacket packet, NetworkRegistry.TargetPoint point);

    void sendToAllTracking(NetworkPacket packet, NetworkRegistry.TargetPoint point);

    void sendToAllTracking(NetworkPacket packet, Entity entity);

    void sendToDimension(NetworkPacket packet, int dimensionId);

    void sendToServer(NetworkPacket packet);

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/network/NetworkHandler.kt
package gregtechlite.gtlitecore.api.network

import net.minecraft.entity.Entity
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraftforge.fml.common.network.NetworkRegistry

interface NetworkHandler
{

    fun registerPacket(packetClass: Class<out NetworkPacket>)

    fun sendToAll(packet: NetworkPacket)

    fun sendTo(packet: NetworkPacket, player: EntityPlayerMP)

    fun sendToAllAround(packet: NetworkPacket, point: NetworkRegistry.TargetPoint)

    fun sendToAllTracking(packet: NetworkPacket, point: NetworkRegistry.TargetPoint)

    fun sendToAllTracking(packet: NetworkPacket, entity: Entity)

    fun sendToDimension(packet: NetworkPacket, dimensionId: Int)

    fun sendToServer(packet: NetworkPacket)
}